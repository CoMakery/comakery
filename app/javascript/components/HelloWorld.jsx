import React from 'react'
import PropTypes from 'prop-types'

export default function HelloWorld(props) {
  return <div>Hello {props.greeting}!</div>
}

HelloWorld.defaultProps = {
  greeting: 'David'
}

HelloWorld.propTypes = {
  greeting: PropTypes.string
}
