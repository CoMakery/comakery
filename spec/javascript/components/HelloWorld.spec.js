import React from 'react'
import HelloWorld from 'components/HelloWorld'
import { shallow } from 'enzyme'

describe('HelloWorld', function() {
  const greeting = 'Hey there'
  let component

  beforeEach(() => {
    component = shallow(<HelloWorld greeting = {greeting}/>)
  })

  it('has text containing props.greeting', () => {
    expect(component.text()).toContain(greeting)
  })

  it('hasn\'t changed', () => {
    expect(component).toMatchSnapshot()
  })
})
