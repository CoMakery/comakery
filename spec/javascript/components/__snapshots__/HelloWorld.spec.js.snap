// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld hasn't changed 1`] = `
<div>
  Hello 
  Hey there
  !
</div>
`;
